CONTRIBUTING.md

```diff
@@ -112,6 +112,18 @@ Core developers should follow these rules when processing pull requests:
 * Use "[Squash and merge](https://github.com/blog/2141-squash-your-commits)"
   to merge PRs.
 * Delete branches for merged PRs (by core devs pushing to the main repo).
+* Edit the final commit message before merging to conform to the following
+  style (we wish to have a clean `git log` output):
+  * When merging a multi-commit PR make sure that the commit message doesn't
+    contain the local history from the committer and the review history from
+    the PR. Edit the message to only describe the end state of the PR.
+  * Make sure there is a *single* newline at the end of the commit message.
```

Comment:
I believe that if you don't provide a newline, the output from `git log` looks like this:

```
Author: David Ross <daboross@daboross.net>
Date:   Fri Jul 21 12:37:47 2017

    Support exported names starting with an underscore in __all__ (#3746)

    Fixes #3745              <--- no empty line after this
commit c4b4949dc471b44b1aa2275d9a564ef01a4ab396
Author: Svyatoslav Ilinskiy <ilinskiy.sv@gmail.com>
Date:   Fri Jul 21 01:02:09 2017
```

I'd prefer this:

```
Author: David Ross <daboross@daboross.net>
Date:   Fri Jul 21 12:37:47 2017

    Support exported names starting with an underscore in __all__ (#3746)

    Fixes #3745
                                 <--- empty line between commits
commit c4b4949dc471b44b1aa2275d9a564ef01a4ab396
Author: Svyatoslav Ilinskiy <ilinskiy.sv@gmail.com>
Date:   Fri Jul 21 01:02:09 2017
```